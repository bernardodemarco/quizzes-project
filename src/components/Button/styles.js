import styled, { css } from 'styled-components';

export const Button = styled.button`
  ${({ theme, background, fontSize, fontWeight, lineHeight }) => css`
    background-color: ${background ? theme.colors.primary : theme.colors.white};
    color: ${background ? theme.colors.white : theme.colors.dark};
    font-size: ${theme.fonts.fontSizes.buttons[fontSize]};
    font-weight: ${theme.fonts.fontWeights[fontWeight]};
    line-height: ${lineHeight};

    font-family: inherit;
    width: 100%;
    height: 5.6rem;
    text-align: center;
    border: none;
    border-radius: 1.6rem;
    padding: 1.6rem 3.2rem;

    &:hover {
      cursor: pointer;
    }
  `}
`;
