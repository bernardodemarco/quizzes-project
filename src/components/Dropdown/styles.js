import styled, { css } from 'styled-components';

export const Dropdown = styled.ul`
  ${({ theme, isVisible }) => css`
    visibility: hidden;
    opacity: 0;

    transition: all 300ms ease-in-out;

    ${
      isVisible &&
      css`
      visibility: visible;
      opacity: 1;
    `
    }

    position: absolute;
    top: 10rem;
    right: 38%;

    width: 14.5rem;
    height: 15.2rem;

    background-color: ${theme.colors.white};
    border-radius: 0.8rem;
    box-shadow: 0rem 0.4rem 0.4rem rgba(0, 0, 0, 0.25);

    list-style: none;

    @media (max-width: 785px) {
      visibility: hidden;
      opacity: 0;
    }
  `}
`;

export const DropdownItem = styled.li`
  ${({ theme }) => css`
    width: 100%;

    padding: 1rem 4.5rem;

    border-bottom: 0.1rem solid ${theme.colors.gray};

    &:last-child {
      border-bottom: none;
    }

    &:hover {
      cursor: pointer;
    }
  `}
`;

export const Text = styled.p`
  ${({ theme }) => css`
    color: #000000;

    text-align: center;

    font-size: ${theme.fonts.fontSizes.paragraphs.large};
    font-weight: ${theme.fonts.fontWeights.normal};
    line-height: 1.8rem;
  `}
`;
