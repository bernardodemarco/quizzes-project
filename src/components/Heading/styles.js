import styled, { css } from 'styled-components';

export const Heading = styled.h1`
  ${({ theme, fontSize, fontWeight, lineHeight }) => css`
    color: ${theme.colors.dark};
    font-size: ${theme.fonts.fontSizes.headings[fontSize]};
    font-weight: ${theme.fonts.fontWeights[fontWeight]};
    line-height: ${lineHeight};
  `}
`;
