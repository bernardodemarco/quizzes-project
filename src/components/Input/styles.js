import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    border: 0.1rem solid ${theme.colors.gray};
    border-radius: 1.2rem;

    width: 100%;
    height: 5.3rem;
    padding: 1.6rem;
    display: flex;
    justify-content: space-between;
    align-items: center;
  `}
`;

export const Input = styled.input`
  ${({ theme }) => css`
    width: 75%;
    border: none;
    font-family: inherit;
    font-weight: ${theme.fonts.fontWeights.light};
    font-size: ${theme.fonts.fontSizes.paragraphs.large};
    color: ${theme.colors.darkGray};
  `}
`;

export const Icon = styled.img`
  ${({ hover = false }) => css`
    width: 3rem;
    height: 2.5rem;

    &:hover {
      cursor: ${hover ? 'pointer' : 'default'};
    }
  `}
`;
