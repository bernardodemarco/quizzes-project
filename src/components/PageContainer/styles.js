import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    width: 100%;
    max-width: 114.6rem;
    min-height: 100vh;

    margin: 0 auto;
    overflow-x: hidden;

    display: flex;
    align-items: center;
  `}
`;
