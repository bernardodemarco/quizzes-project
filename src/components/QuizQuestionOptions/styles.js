import styled, { css } from 'styled-components';
import { Button } from '../Button/styles';

export const OptionsContainer = styled.div`
  ${({ theme, finished }) => css`
    width: 100%;

    display: flex;
    flex-direction: column;
    gap: 2rem;

    & .correct-answer, & .wrong-answer {
      background-color: ${theme.colors.lightGray};
    }

    & .correct-answer {
      border: 0.2rem solid ${theme.colors.success};
    }

    & .wrong-answer {
      border: 0.2rem solid ${theme.colors.error};
    }

    & div:hover,
    & div input:hover,
    & div label:hover,
    & div label span:hover {
      cursor: ${finished ? 'not-allowed' : 'pointer'}
    }

    & ${Button} {
      margin-top: 2rem;
      line-height: 1.8rem;
    }
  `}
`;

export const OptionWrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;

    border: 0.1rem solid ${theme.colors.gray};
    border-radius: 0.8rem;

    padding: 1.6rem 2.4rem;
  `}
`;

export const Label = styled.label`
  ${({ theme }) => css`
    span {
      margin-right: 1.6rem;

      font-weight: ${theme.fonts.fontWeights.light};
      font-size: ${theme.fonts.fontSizes.paragraphs.large};
      line-height: 2.6rem;

      color: ${theme.colors.dark};
    }
  `}
`;

export const RadioInput = styled.input`
  ${() => css`
    visibility: hidden;
    width: 0;
    height: 0;
  `}
`;
