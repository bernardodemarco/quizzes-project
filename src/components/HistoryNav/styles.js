import styled, { css } from 'styled-components';

import { Container as PageContainer } from './../PageContainer/styles';
import { ImageContainer as ReturnButton } from './../ReturnButton/styles';
import { Heading } from './../Heading/styles';

export const Nav = styled.nav`
  ${({ theme }) => css`
    z-index: 1;
    position: fixed;
    top: 0;
    right: 0;
    left: 0;

    height: 12rem;
    border-bottom: 0.1rem solid ${theme.colors.gray};

    ${PageContainer} {
      min-height: initial;
      height: 100%;
      justify-content: flex-start;
      align-items: center;
    }

    ${Heading} {
      z-index: 1;
      position: fixed;
      left: 14.7rem;
    }

    @media ${theme.media.mediumDevices} {
      ${ReturnButton} {
        left: 5rem;
      }
    }

    @media ${theme.media.smallDevices} {
      ${Heading} {
        left: 12rem;
        font-size: ${theme.fonts.fontSizes.headings.medium};
      }
    }
  `}
`;
