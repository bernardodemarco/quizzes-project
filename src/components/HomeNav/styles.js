import styled, { css } from 'styled-components';

import { Container as PageContainer } from '../PageContainer/styles';
import { Input, Container as InputContainer } from '../Input/styles';
import { Heading } from '../Heading/styles';

export const Button = styled.button`
  ${({ theme }) => css`
    display: none;

    z-index: 2;
    position: fixed;
    top: 3rem;
    right: 3rem;

    height: 4rem;
    width: 4rem;

    color: ${theme.colors.primary};
    background-color: ${theme.colors.lightGray};
    border: none;

    &:hover {
      cursor: pointer;
    }

    @media (max-width: 785px) {
      display: block;
    }
  `}
`;

export const Container = styled.div`
  ${({ theme, isVisible }) => css`
    height: 12rem;

    z-index: 1;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;

    border-bottom: 0.1rem solid ${theme.colors.gray};

    transition: all 300ms ease-in-out;

    & ${PageContainer} {
      min-height: initial;
      height: 100%;
    }

    @media (max-width: 1300px) {
      & ${PageContainer} {
        padding: 0 3rem;
      }
    }

    @media (max-width: 785px) {
      & {
        height: 100vh;
        bottom: 0;
        border-bottom: none;

        visibility: hidden;
        opacity: 0;
        ${
          isVisible &&
          css`
          visibility: visible;
          opacity: 1;
        `
        }
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${() => css`
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: flex-end;

    @media (max-width: 785px) {
      & {
        height: 100%;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        gap: 7.5rem;
      }
    }
  `}
`;

export const Logo = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    align-items: flex-start;

    & ${Heading} {
      text-align: center;
    }

    @media (max-width: 1130px) {
      & ${Heading} {
        font-size: ${theme.fonts.fontSizes.headings.medium};
        margin-right: 2rem;
      }
    }

    @media ${theme.media.mediumDevices} {
      & ${Heading} {
        font-size: ${theme.fonts.fontSizes.headings.small};
      }
    }

    @media (max-width: 785px) {
      & {
        justify-content: center;
        align-items: center;
      }

      & ${Heading} {
        margin-right: 0;
      }
    }
  `}
`;

export const Nav = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-direction: row;
    align-items: flex-end;
    gap: 3rem;

    & ${Input} {
      width: 40.4rem;
    }

    @media (max-width: 1130px) {
      & {
        align-items: center;
      }

      & ${Input} {
        width: 30rem;
      }
    }

    @media ${theme.media.mediumDevices} {
      & ${Input} {
        width: 22.5rem;
      }
    }

    @media (max-width: 785px) {
      & {
        flex-direction: column;
        justify-content: center;
        align-items: center;
      }

      & ${InputContainer}, & ${Input}, & .dropdown-button  {
        display: none;
      }
    }
  `}
`;

export const Link = styled.a`
  ${({ theme }) => css`
    display: block;
    text-align: center;
    color: ${theme.colors.dark};

    font-weight: ${theme.fonts.fontWeights.light};
    font-size: ${theme.fonts.fontSizes.paragraphs.large};
    line-height: 2.1rem;

    &:hover {
      cursor: pointer;
    }

    @media (max-width: 785px) {
      & {
        font-size: 1.8rem;
      }
    }
  `}
`;

export const Theme = styled.p`
  ${({ theme }) => css`
    display: none;
    color: ${theme.colors.dark};
    text-align: center;

    font-weight: ${theme.fonts.fontWeights.light};
    font-size: ${theme.fonts.fontSizes.paragraphs.large};
    line-height: 2.1rem;

    &:hover {
      cursor: pointer;
    }

    @media (max-width: 785px) {
      & {
        display: block;
        font-size: 1.8rem;
      }
    }
  `}
`;
