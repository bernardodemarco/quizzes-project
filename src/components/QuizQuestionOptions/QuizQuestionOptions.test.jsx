import { renderThemeProvider } from './../../styles/render-theme-provider';
import { fireEvent, screen } from '@testing-library/react';

import { QuizQuestionOptions } from '.';
import data from './mock';

describe('<QuizQuestionOptions />', () => {
  it('should render <QuizQuestionOptions /> correctly', () => {
    const func = jest.fn();
    renderThemeProvider(<QuizQuestionOptions {...data} onButtonClick={func} />);

    expect(screen.getAllByRole('radio', { hidden: true })).toHaveLength(
      data.answers.length,
    );

    expect(
      screen.queryByRole('button', { name: 'Continuar' }),
    ).not.toBeInTheDocument();
  });

  it('should apply the correct styles when the wrong option has been chosen', () => {
    const func = jest.fn();
    renderThemeProvider(<QuizQuestionOptions {...data} onButtonClick={func} />);

    const radios = screen.getAllByRole('radio', { hidden: true });
    const optionWrappers = radios.map((radio) => {
      return radio.parentElement;
    });

    for (let radio of radios) {
      expect(radio).not.toHaveAttribute('disabled', '');
    }

    expect(
      screen.queryByRole('button', { name: 'Continuar' }),
    ).not.toBeInTheDocument();

    const selectedElement = optionWrappers[data.correctAnswerIndex - 1];
    const initialClassName = selectedElement.getAttribute('class');

    fireEvent.click(selectedElement);

    expect(selectedElement).toHaveAttribute(
      'class',
      initialClassName + ' wrong-answer',
    );

    for (let radio of radios) {
      expect(radio).toHaveAttribute('disabled', '');
    }

    expect(
      screen.queryByRole('button', { name: 'Continuar' }),
    ).toBeInTheDocument();

    // everything will remain the same when clicked again
    fireEvent.click(selectedElement);
    expect(
      screen.queryByRole('button', { name: 'Continuar' }),
    ).toBeInTheDocument();
  });

  it('should apply the correct styles when the correct option has been chosen', () => {
    const func = jest.fn();
    renderThemeProvider(<QuizQuestionOptions {...data} onButtonClick={func} />);

    const radios = screen.getAllByRole('radio', { hidden: true });
    const optionWrappers = radios.map((radio) => {
      return radio.parentElement;
    });

    const selectedElement = optionWrappers[data.correctAnswerIndex];
    const initialClassName = selectedElement.getAttribute('class');

    fireEvent.click(selectedElement);

    expect(selectedElement).toHaveAttribute(
      'class',
      initialClassName + ' correct-answer',
    );
  });

  it('should change class attribute in OptionWrapper if input or label were clicked', () => {
    const func = jest.fn();
    renderThemeProvider(<QuizQuestionOptions {...data} onButtonClick={func} />);

    const radios = screen.getAllByRole('radio', { hidden: true });

    // clicking on <label />
    const selectedElement =
      radios[data.correctAnswerIndex - 1].nextElementSibling;
    const affectedElement = selectedElement.parentElement;
    const initialClassName = affectedElement.getAttribute('class');

    fireEvent.click(selectedElement);

    expect(affectedElement).toHaveAttribute(
      'class',
      initialClassName + ' wrong-answer',
    );
  });

  it('should change class attribute in OptionWrapper if span was clicked', () => {
    const func = jest.fn();
    renderThemeProvider(<QuizQuestionOptions {...data} onButtonClick={func} />);

    const radios = screen.getAllByRole('radio', { hidden: true });

    // clicking on <span />
    const selectedElement =
      radios[data.correctAnswerIndex].nextSibling.firstChild;
    const affectedElement = selectedElement.parentElement.parentElement;
    const initialClassName = affectedElement.getAttribute('class');

    fireEvent.click(selectedElement);

    expect(affectedElement).toHaveAttribute(
      'class',
      initialClassName + ' correct-answer',
    );
  });

  it('should match snapshot', () => {
    const func = jest.fn();
    const { container } = renderThemeProvider(
      <QuizQuestionOptions {...data} onButtonClick={func} />,
    );
    expect(container).toMatchSnapshot();
  });
});
