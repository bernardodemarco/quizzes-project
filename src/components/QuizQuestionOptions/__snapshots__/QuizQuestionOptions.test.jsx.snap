// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuizQuestionOptions /> should match snapshot 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 2rem;
}

.c0 .correct-answer,
.c0 .wrong-answer {
  background-color: #FFF5EE;
}

.c0 .correct-answer {
  border: 0.2rem solid #5BA092;
}

.c0 .wrong-answer {
  border: 0.2rem solid #EF4949;
}

.c0 div:hover,
.c0 div input:hover,
.c0 div label:hover,
.c0 div label span:hover {
  cursor: pointer;
}

.c1 {
  width: 100%;
  border: 0.1rem solid #BEBAB3;
  border-radius: 0.8rem;
  padding: 1.6rem 2.4rem;
}

.c3 span {
  margin-right: 1.6rem;
  font-weight: 400;
  font-size: 1.6rem;
  line-height: 2.6rem;
  color: #3C3A36;
}

.c2 {
  visibility: hidden;
  width: 0;
  height: 0;
}

<div>
  <div
    class="c0"
    id="options-container"
  >
    <div
      class="c1"
      id="0"
    >
      <input
        class="c2"
        id="answer-0"
        name="quiz-option"
        type="radio"
      />
      <label
        class="c3"
        for="answer-0"
      >
        <span>
          A.
        </span>
        <span>
          Resposta 01
        </span>
      </label>
    </div>
    <div
      class="c1"
      id="1"
    >
      <input
        class="c2"
        id="answer-1"
        name="quiz-option"
        type="radio"
      />
      <label
        class="c3"
        for="answer-1"
      >
        <span>
          B.
        </span>
        <span>
          Resposta 02
        </span>
      </label>
    </div>
    <div
      class="c1"
      id="2"
    >
      <input
        class="c2"
        id="answer-2"
        name="quiz-option"
        type="radio"
      />
      <label
        class="c3"
        for="answer-2"
      >
        <span>
          C.
        </span>
        <span>
          Resposta 03
        </span>
      </label>
    </div>
  </div>
</div>
`;
